combustion-android-ble/src/main/java/inc/combustion/framework/ble/uart/meatnet/NodeRequest.kt

```diff
@@ -0,0 +1,224 @@
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package inc.combustion.framework.ble.uart.meatnet
+
+import inc.combustion.framework.ble.*
+import inc.combustion.framework.ble.getCRC16CCITT
+import inc.combustion.framework.ble.getLittleEndianUInt32At
+import inc.combustion.framework.ble.putLittleEndianUInt32At
+import inc.combustion.framework.ble.putLittleEndianUShortAt
+
+
+/**
+ * Baseclass for UART request messages
+ */
+internal open class NodeRequest() : NodeUARTMessage() {
+    companion object {
+        const val HEADER_SIZE: UByte = 10u
+
+        /**
+         * Factory method for parsing a request from data.
+         *
+         * @param data Raw data to parse
+         * @return Instant of request if found or null if one could not be parsed from the data
+         */
+        fun requestFromData(data : UByteArray) : NodeRequest? {
+            // Sync bytes
+            val syncBytes = data.slice(0..1)
+            val syncString = String(syncBytes.toUByteArray().toByteArray())
+            if(syncString != "cafe") {
+                print("NodeRequest::fromData(): Missing sync bytes in request")
+                return null
+            }
+
+            // Message type
+            val typeByte = data[4]
+
+            val messageType = NodeMessageType.fromUByte(typeByte)
+            if(messageType == null) {
+                print("NodeRequest::fromData(): Unknown message type in request")
+                return null
+            }
+
+            // Request ID
+            val requestId = data.getLittleEndianUInt32At(5)
+
+            // Payload Length
+            val payloadLength = data[9]
+
+            // CRC
+            val crc = data.getLittleEndianUShortAt(2)
+
+            val crcDataLength = 6 + payloadLength.toInt()
+
+            var crcData = data.drop(4).toUByteArray()
+            crcData = crcData.dropLast(crcData.size - crcDataLength).toUByteArray()
+
+            val calculatedCRC = crcData.getCRC16CCITT()
+
+            if(crc != calculatedCRC) {
+                print("NodeRequest::fromData(): Invalid CRC")
+                return null
+            }
+
+            val requestLength = payloadLength.toInt() + HEADER_SIZE.toInt()
+
+            // Invalid number of bytes
+            if(data.size < requestLength) {
+                return null
+            }
+
+            // print("Success: \(success), payloadLength: \(payloadLength)")
```

Comment:
Can you remove any other commented out code?

Reply:
Done.